Source/src/modules/ModuleDebugMode.cpp

```diff
@@ -43,6 +52,64 @@ UpdateStatus Hachiko::ModuleDebugMode::Update(const float delta)
 
 	RenderGui();
 
+
+	// God mode
+	
+	// Teleport player with mouse double click
+	if (App->input->GetKey(SDL_SCANCODE_RALT) == Hachiko::KeyState::KEY_REPEAT && ImGui::IsMouseDoubleClicked(0))
+	{
+        Plane plane(vec(50, 0, 50), vec(50, 0, -50), vec(-50, 0, -50));
+        float d = 0;
+        if (line.Intersects(plane, &d))
+        {
+            player = FindPlayer();
+            player->GetTransform()->SetGlobalPosition(line.GetPoint(d));
+        }
+	}
+
+	// Teleport player to the nearest position in playerLocations
+	if (App->input->GetKey(SDL_SCANCODE_SPACE) == Hachiko::KeyState::KEY_DOWN)
+	{
+        player = FindPlayer();
+        float3 currentPosition = player->GetTransform()->GetGlobalPosition();
+		
+		float dist = 1000.0f;
+		int loc = 0;
+		for (int i = 0; i < playerLocations.size(); ++i)
+		{
+            if (dist > math::Distance(currentPosition, playerLocations[i]) && math::Distance(currentPosition, playerLocations[i]) != 0)
+			{
+                dist = math::Distance(currentPosition, playerLocations[i]);
+                loc = i;
+			}
+		}
+        player->GetTransform()->SetGlobalPosition(playerLocations[loc]);
+	}
+	
+	// Teleport player to the next position in playerLocations
+    if (App->input->GetKey(SDL_SCANCODE_RETURN) == Hachiko::KeyState::KEY_DOWN)
+	{
+        player = FindPlayer();
+        float3 currentPosition = player->GetTransform()->GetGlobalPosition();
+        
+		int loc = 0;
+		for (int i = 0; i < playerLocations.size(); ++i)
+		{
+            if (math::Distance(currentPosition, playerLocations[i]) == 0)
+			{
+                loc = i + 1;
+                break;
+			}
+		}
+
+		if (loc == playerLocations.size())
+		{
+            loc = 0;
+		}
+
+        player->GetTransform()->SetGlobalPosition(playerLocations[loc]);
+	}
+
```

Comment:
For the sake of readability of Update, I suggest to move the content of each if statement to private functions and the use of else if to avoid unnecesary checks. E.g:

if (App->input->GetKey(SDL_SCANCODE_RALT) == Hachiko::KeyState::KEY_REPEAT && ImGui::IsMouseDoubleClicked(0))
{
    TeleportMouseDoubleClickPosition()
}
else if (App->input->GetKey(SDL_SCANCODE_SPACE) == Hachiko::KeyState::KEY_DOWN)
{
    TeleportNearestPosition()
}
else if (App->input->GetKey(SDL_SCANCODE_RETURN) == Hachiko::KeyState::KEY_DOWN)
{
    TeleportNextPlayerPosition()
}